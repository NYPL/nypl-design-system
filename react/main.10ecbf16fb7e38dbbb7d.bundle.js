(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{1292:function(n,o,c){"use strict";c.r(o);c(1293),c(1323),c(1335),c(1442),c(1485)},1448:function(n,o){},1449:function(n,o){},1450:function(n,o){},1451:function(n,o){},501:function(n,o,c){c(502),c(645),n.exports=c(1292)},564:function(n,o){}},[[501,1,2]]]);