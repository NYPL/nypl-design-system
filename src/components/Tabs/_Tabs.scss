.reusable-tabs {
  margin: 20px;
  padding: 0;

  .tab-navs {
    list-style: none;
    margin: 0;
    padding: 0;

    .tab-nav {
      display: inline-block;

      &.current {
        button {
          background: var(--ui-gray-xdark);
          color: var(--ui-white);
        }
      }

      .btn {
        background: var(--ui-gray-medium);
        color: var(--ui-gray-xdark);
        transition: background 0.5s, color 0.5s;

        &:hover {
          background: var(--ui-gray-xdark);
          color: var(--ui-white);
        }

        &:active,
        &:focus {
          color: var(--ui-white);
        }
      }
    }
  }

  .tab-content {
    border: 1px solid var(--ui-gray-xdark);
    padding: 5px;
  }

  .hidden {
    display: none;
  }
}
