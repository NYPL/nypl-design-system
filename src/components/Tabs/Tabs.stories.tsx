import * as React from "react";
import Tabs from "./Tabs";

export default {
    title: "Tabs",
    component: Tabs,
};

const tabItems = {
    "1": <p>Some content for the first tab item...</p>,
    "2": <p>Content for the second tab item...</p>,
    "3": <p>...And content for the third tab item!</p>,
};

export const tabs = () => <Tabs items={tabItems} id="demo" />;

tabs.storyName = "Tabs";
