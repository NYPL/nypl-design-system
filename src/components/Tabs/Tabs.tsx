import * as React from "react";
import {
  Box,
  Tab,
  TabList,
  TabPanels,
  TabPanel,
  Tabs as ChakraTabs,
  useMultiStyleConfig,
} from "@chakra-ui/react";

import generateUUID from "../../helpers/generateUUID";
import { IconNames, IconRotationTypes, IconSizes } from "../Icons/IconTypes";
import Icon from "../Icons/Icon";
import { ButtonTypes } from "../Button/ButtonTypes";
import Button from "../Button/Button";
import useCarouselStyles from "../../hooks/useCarouselStyles";
import useWindowSize from "../../hooks/useWindowSize";

// Internal interface used for rendering `Tabs` tab and panel
// elements, either from data or from children.
interface TabPanelProps {
  tabs: React.ReactNode[] | React.ReactNode;
  panels: React.ReactNode[] | React.ReactNode;
}
// The general shape of the data object for each Tab.
export interface TabsContentDataProps {
  label: string;
  content: string | React.ReactNode;
}

export interface TabsProps {
  /** Array of data to display */
  contentData?: TabsContentDataProps[];
  /** The index of the tab to display for controlled situations. */
  defaultIndex?: number;
  /** ID that other components can cross reference for accessibility purposes */
  id?: string;
  /** The callback function invoked on every tab change event. */
  onChange?: (index: number) => any;
  /** Render a hash in the url for each tab. Only available when data is
   * passed through the `data` props. */
  useHash?: boolean;
}

/**
 * An internal function used to update the hash in the URL.
 * This function is only used when `useHash` is `true`.
 */
const onClickHash = (tabHash) => {
  window.location.hash = tabHash;
};

/**
 * This returns an object with `Tab` and `TabPanel` components to rendered in
 * `TabList` and `TabPanels` components respectively.
 */
const getElementsFromContentData = (data, useHash): TabPanelProps => {
  const tabs = [];
  const panels = [];

  if (!data?.length) {
    return { tabs: [], panels: [] };
  }

  if (data?.length > 6) {
    console.warn(
      "NYPL Reservoir Tabs: it is recommended to use no more than six tabs. If " +
        "more than six tabs are needed, consider other navigational patterns."
    );
  }
  data.forEach((tab, index) => {
    let tempPanel;
    // For URL hash enabled tabs, we need to add a custom `onClick` to handle the URL hash.
    const tempTab = (
      <Tab
        fontSize={["0", null, "1"]}
        key={index}
        onClick={useHash ? () => onClickHash(`tab${index + 1}`) : null}
      >
        {tab.label}
      </Tab>
    );
    if (typeof tab.content === "string") {
      tempPanel = (
        <TabPanel
          dangerouslySetInnerHTML={{ __html: tab.content }}
          key={index}
        />
      );
    } else {
      tempPanel = <TabPanel key={index}>{tab.content}</TabPanel>;
    }

    tabs.push(tempTab);
    panels.push(tempPanel);
  });

  return {
    tabs: [<TabList key="list">{tabs}</TabList>],
    panels: [<TabPanels key="panels">{panels}</TabPanels>],
  };
};

/**
 * This returns an object with `TabList` and `TabPanels` components to help format
 * the DOM when building up the `Tabs` component using child component.
 */
const getElementsFromChildren = (children): TabPanelProps => {
  const tabs = [];
  const panels = [];

  if (!children?.length) {
    return { tabs: [], panels: [] };
  }

  children.forEach((child) => {
    if (child.type === TabList || child.props.mdxType === "TabList") {
      tabs.push(child);

      const childTabs = React.Children.count(child.props.children);
      if (childTabs > 6) {
        console.warn(
          "NYPL Reservoir Tabs: It is recommended to use no more than six tabs. " +
            "If more than six tabs are needed, consider other navigational patterns."
        );
      }
    }

    if (child.type === TabPanels || child.props.mdxType === "TabPanels") {
      panels.push(child);
    }
  });

  return { tabs, panels };
};

/**
 * Renders Chakra's `Tab` component with specific variants, props,
 * and controlled styling.
 */
function Tabs(props: React.PropsWithChildren<TabsProps>) {
  const {
    children,
    contentData,
    defaultIndex = 0,
    id = generateUUID(),
    onChange,
    useHash = false,
  } = props;
  const styles = useMultiStyleConfig("Tabs", {});
  // Just an estimate of the tab width for the mobile carousel.
  const initTabWidth = 65;
  // An estimate for the tab width for larger device widths.
  const mediumTabWidth = 40;
  const [tabWidth, setTabWidth] = React.useState(initTabWidth);
  const windowDimensions = useWindowSize();
  const { tabs, panels }: any = contentData
    ? getElementsFromContentData(contentData, useHash)
    : getElementsFromChildren(children);

  if (tabs.length === 0 || panels.length === 0) {
    console.warn(
      "NYPL Reservoir Tabs: Pass data in the `contentData` props or as children."
    );
  }

  // `tabs` is an array for the children component approach but an object for
  // the `contentData` prop approach. We need to get the right props key value
  // to set the carousel's length.
  const tabProps = tabs[0] ? tabs[0]?.props : (tabs as any).props;
  const { prevSlide, nextSlide, carouselStyle, goToStart } = useCarouselStyles(
    tabProps?.children?.length,
    tabWidth
  );
  React.useEffect(() => {
    if (windowDimensions.width > 320) {
      setTabWidth(mediumTabWidth);
    } else {
      setTabWidth(initTabWidth);
    }
    // If we are on larger viewports, reset the carousel so all tabs display.
    if (windowDimensions.width > 600) {
      goToStart();
    }
  }, [goToStart, windowDimensions.width]);
  const previousButton = (
    <Button
      attributes={{
        "aria-label": "Previous",
        ...styles.buttonArrows,
        left: "0",
      }}
      buttonType={ButtonTypes.Primary}
      id={`tabs-previous-${id}`}
      onClick={prevSlide}
    >
      <Icon
        iconRotation={IconRotationTypes.Rotate90}
        id={`tabs-previous-icon-${id}`}
        name={IconNames.Arrow}
        size={IconSizes.Small}
      />
    </Button>
  );
  const nextButton = (
    <Button
      attributes={{
        "aria-label": "Next",
        ...styles.buttonArrows,
        right: "0",
      }}
      buttonType={ButtonTypes.Primary}
      id={`tabs-next-${id}`}
      onClick={nextSlide}
    >
      <Icon
        iconRotation={IconRotationTypes.Rotate270}
        id={`tabs-next-icon-${id}`}
        name={IconNames.Arrow}
        size={IconSizes.Small}
      />
    </Button>
  );

  if (children && contentData?.length) {
    console.warn(
      "NYPL Reservoir Tabs: Only pass children or data in the `contentData` " +
        "prop. Do not pass both."
    );
  }

  return (
    <ChakraTabs
      defaultIndex={defaultIndex}
      id={id}
      // The following lazy loads each panel whenever it is needed.
      isLazy
      onChange={onChange}
      variant="enclosed"
    >
      <Box
        __css={styles.tablistWrapper}
        sx={{
          "&::-webkit-scrollbar": {
            display: "none",
          },
        }}
      >
        {previousButton}
        <Box __css={styles.carouselParent}>
          <Box {...carouselStyle}>{tabs}</Box>
        </Box>
        {nextButton}
      </Box>
      {panels}
    </ChakraTabs>
  );
}

export { Tabs, TabList, Tab, TabPanels, TabPanel };
