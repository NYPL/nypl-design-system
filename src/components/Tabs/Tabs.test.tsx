import * as React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

import { Tabs, TabList, Tab, TabPanels, TabPanel } from "./Tabs";

export const animalCrossing = [
  {
    label: "Tom Nook",
    content: (
      <p>
        Tom Nook, <b>known in Japan as Tanukichi</b>, is a fictional character
        in the Animal Crossing series who operates the village store.
      </p>
    ),
  },
  {
    label: "Isabelle",
    content:
      "Isabelle, known as Shizue in Japan, is a fictional character " +
      "from the Animal Crossing series of video games. She is a kindly Shih " +
      "Tzu that debuted in the 2012 release Animal Crossing: New Leaf, where " +
      "she serves as the secretary to the player character.",
  },
  {
    label: "K.K. Slider",
    content:
      "<p>Totakeke, more commonly known as <b>K.K. Slider or K.K.</b>, is a " +
      "fictional character within the Animal Crossing franchise. One of the " +
      "franchise's most popular characters, he debuted in the title Animal " +
      "Crossing, and has appeared in every installment since.</p>",
  },
];

describe("Tabs Accessibility", () => {
  it("passes axe accessibility test with the data prop", async () => {
    const { container } = render(<Tabs tabsData={animalCrossing} />);
    expect(await axe(container)).toHaveNoViolations();
  });

  it("passes axe accessibility test with children components", async () => {
    const { container } = render(
      <Tabs>
        <TabList>
          <Tab>Tom Nook</Tab>
          <Tab>Isabelle</Tab>
          <Tab>K.K. Slider</Tab>
        </TabList>
        <TabPanels>
          <TabPanel>
            Tom Nook, known in Japan as Tanukichi, is a fictional character in
            the Animal Crossing series who operates the village store.
          </TabPanel>
          <TabPanel>
            Isabelle, known as Shizue in Japan, is a fictional character from
            the Animal Crossing series of video games. She is a kindly Shih Tzu
            that debuted in the 2012 release Animal Crossing: New Leaf, where
            she serves as the secretary to the player character.
          </TabPanel>
          <TabPanel>
            Totakeke, more commonly known as K.K. Slider or K.K., is a fictional
            character within the Animal Crossing franchise. One of the
            franchise's most popular characters, he debuted in the title Animal
            Crossing, and has appeared in every installment since.
          </TabPanel>
        </TabPanels>
      </Tabs>
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("Tabs", () => {
  const getTabByName = (name: string) => screen.getByRole("tab", { name });

  it("renders all tabs but only one panel at a time with children", () => {
    render(
      <Tabs>
        <TabList>
          <Tab>Tom Nook</Tab>
          <Tab>Isabelle</Tab>
          <Tab>K.K. Slider</Tab>
        </TabList>
        <TabPanels>
          <TabPanel>
            Tom Nook, known in Japan as Tanukichi, is a fictional character in
            the Animal Crossing series who operates the village store.
          </TabPanel>
          <TabPanel>
            Isabelle, known as Shizue in Japan, is a fictional character from
            the Animal Crossing series of video games. She is a kindly Shih Tzu
            that debuted in the 2012 release Animal Crossing: New Leaf, where
            she serves as the secretary to the player character.
          </TabPanel>
          <TabPanel>
            Totakeke, more commonly known as K.K. Slider or K.K., is a fictional
            character within the Animal Crossing franchise. One of the
            franchise's most popular characters, he debuted in the title Animal
            Crossing, and has appeared in every installment since.
          </TabPanel>
        </TabPanels>
      </Tabs>
    );

    expect(getTabByName("Tom Nook")).toBeInTheDocument();
    expect(getTabByName("Isabelle")).toBeInTheDocument();
    expect(getTabByName("K.K. Slider")).toBeInTheDocument();
    expect(
      screen.getByText(/Tom Nook, known in Japan as Tanukichi/i)
    ).toBeInTheDocument();
    expect(
      screen.queryByText(/Isabelle, known as Shizue in Japan,/i)
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText(
        /Totakeke, more commonly known as K.K. Slider or K.K./i
      )
    ).not.toBeInTheDocument();
  });

  it("renders all tabs but only one visible panel at a time with data prop", () => {
    render(<Tabs tabsData={animalCrossing} />);
    expect(getTabByName("Tom Nook")).toBeInTheDocument();
    expect(getTabByName("Isabelle")).toBeInTheDocument();
    expect(getTabByName("K.K. Slider")).toBeInTheDocument();
    expect(
      screen.getByText(/known in Japan as Tanukichi/i)
    ).toBeInTheDocument();
    expect(
      screen.queryByText(/Isabelle, known as Shizue in Japan,/i)
    ).not.toBeVisible();
    expect(
      screen.queryByText(/Totakeke, more commonly known as/i)
    ).not.toBeVisible();
  });

  it("switches between tabs", () => {
    render(<Tabs tabsData={animalCrossing} />);
    const isabelleTab = getTabByName("Isabelle");
    const kkSliderTab = getTabByName("K.K. Slider");

    expect(
      screen.getByText(/known in Japan as Tanukichi/i)
    ).toBeInTheDocument();
    expect(
      screen.queryByText(/Isabelle, known as Shizue in Japan,/i)
    ).not.toBeVisible();
    expect(
      screen.queryByText(/Totakeke, more commonly known as/i)
    ).not.toBeVisible();

    userEvent.click(isabelleTab);
    expect(isabelleTab).toHaveAttribute("aria-selected", "true");

    expect(
      screen.queryByText(/known in Japan as Tanukichi/i)
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText(/Isabelle, known as Shizue in Japan,/i)
    ).toBeInTheDocument();
    expect(
      screen.queryByText(/Totakeke, more commonly known/i)
    ).not.toBeVisible();

    userEvent.click(kkSliderTab);
    expect(kkSliderTab).toHaveAttribute("aria-selected", "true");

    expect(
      screen.queryByText(/known in Japan as Tanukichi/i)
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText(/Isabelle, known as Shizue in Japan,/i)
    ).not.toBeVisible();
    expect(
      screen.queryByText(/Totakeke, more commonly known/i)
    ).toBeInTheDocument();
  });

  it("renders the specified initial index value", () => {
    render(<Tabs tabsData={animalCrossing} defaultIndex={2} />);
    let tomTab = getTabByName("Tom Nook");
    let isabelleTab = getTabByName("Isabelle");
    let kkSliderTab = getTabByName("K.K. Slider");

    expect(tomTab).toHaveAttribute("aria-selected", "false");
    expect(isabelleTab).toHaveAttribute("aria-selected", "false");
    expect(kkSliderTab).toHaveAttribute("aria-selected", "true");
  });

  it("invokes the callback function", () => {
    let selectedIndex = 0;
    const onChange = (index: number) => (selectedIndex = index);

    render(<Tabs tabsData={animalCrossing} onChange={onChange} />);

    const tomTab = getTabByName("Tom Nook");
    const isabelleTab = getTabByName("Isabelle");
    const kkSliderTab = getTabByName("K.K. Slider");

    userEvent.click(kkSliderTab);
    expect(selectedIndex).toEqual(2);

    userEvent.click(tomTab);
    expect(selectedIndex).toEqual(0);

    userEvent.click(isabelleTab);
    expect(selectedIndex).toEqual(1);
  });

  it("should throw a warning when no data is passed", () => {
    const warn = jest.spyOn(console, "warn");
    render(<Tabs />);
    expect(warn).toHaveBeenCalledWith(
      "NYPL Reservoir Tabs: Pass data in the `contentData` props or as children."
    );
  });

  it("should throw a warning when both the 'data' prop and children are passed", () => {
    const warn = jest.spyOn(console, "warn");
    render(
      <Tabs tabsData={animalCrossing}>
        <TabList>
          <Tab>Tom Nook</Tab>
          <Tab>Isabelle</Tab>
          <Tab>K.K. Slider</Tab>
        </TabList>
        <TabPanels>
          <TabPanel>
            Tom Nook, known in Japan as Tanukichi, is a fictional character in
            the Animal Crossing series who operates the village store.
          </TabPanel>
          <TabPanel>
            Isabelle, known as Shizue in Japan, is a fictional character from
            the Animal Crossing series of video games. She is a kindly Shih Tzu
            that debuted in the 2012 release Animal Crossing: New Leaf, where
            she serves as the secretary to the player character.
          </TabPanel>
          <TabPanel>
            Totakeke, more commonly known as K.K. Slider or K.K., is a fictional
            character within the Animal Crossing franchise. One of the
            franchise's most popular characters, he debuted in the title Animal
            Crossing, and has appeared in every installment since.
          </TabPanel>
        </TabPanels>
      </Tabs>
    );
    expect(warn).toHaveBeenCalledWith(
      "NYPL Reservoir Tabs: Only pass children or data in the `contentData` " +
        "prop. Do not pass both."
    );
  });

  it("should throw a warning when more than six object tabs are passed", () => {
    const warn = jest.spyOn(console, "warn");
    render(
      <Tabs
        tabsData={[
          ...animalCrossing,
          ...[
            { label: "Another character 1", content: "Text" },
            { label: "Another character 2", content: "Text" },
            { label: "Another character 3", content: "Text" },
            { label: "Another character 4", content: "Text" },
          ],
        ]}
      />
    );
    expect(warn).toHaveBeenCalledWith(
      "NYPL Reservoir Tabs: it is recommended to use no more than six tabs. If " +
        "more than six tabs are needed, consider other navigational patterns."
    );
  });

  it("renders the UI snapshot correctly", () => {
    const basic = renderer
      .create(<Tabs tabsData={animalCrossing} id="basic" />)
      .toJSON();
    const withChakraProps = renderer
      .create(
        <Tabs
          tabsData={animalCrossing}
          id="chakra"
          p="20px"
          color="ui.error.primary"
        />
      )
      .toJSON();
    const withOtherProps = renderer
      .create(<Tabs tabsData={animalCrossing} id="props" data-testid="props" />)
      .toJSON();

    expect(basic).toMatchSnapshot();
    expect(withChakraProps).toMatchSnapshot();
    expect(withOtherProps).toMatchSnapshot();
  });
});
