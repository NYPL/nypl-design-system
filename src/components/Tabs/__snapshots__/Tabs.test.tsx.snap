// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders the UI snapshot correctly 1`] = `
<div
  className="chakra-tabs css-1xdhyk6"
  id="basic"
>
  <div
    className="css-l3f8vc"
  >
    <button
      aria-label="Scroll tabs left"
      className="chakra-button css-18qhlcl"
      id="tabs-previous-basic"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="tabs-previous-icon-basic"
        role="img"
        title="previous arrow"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="css-0"
    >
      <div
        className="css-tjdjds"
      >
        <div
          aria-orientation="horizontal"
          className="chakra-tabs__tablist css-k008qs"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="basic--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="basic--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Tom Nook
          </button>
          <button
            aria-controls="basic--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="basic--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Isabelle
          </button>
          <button
            aria-controls="basic--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="basic--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            K.K. Slider
          </button>
        </div>
      </div>
    </div>
    <button
      aria-label="Scroll tabs right"
      className="chakra-button css-1djleqc"
      id="tabs-next-basic"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="tabs-next-icon-basic"
        role="img"
        title="next arrow"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
  </div>
  <div
    className="chakra-tabs__tab-panels css-8atqhb"
  >
    <div
      aria-labelledby="basic--tab-0"
      className="chakra-tabs__tab-panel css-sjt5nk"
      hidden={false}
      id="basic--tabpanel-0"
      role="tabpanel"
      tabIndex={0}
    >
      <p>
        Tom Nook, 
        <b>
          known in Japan as Tanukichi
        </b>
        , is a fictional character in the Animal Crossing series who operates the village store.
      </p>
    </div>
    <div
      aria-labelledby="basic--tab-1"
      className="chakra-tabs__tab-panel css-sjt5nk"
      dangerouslySetInnerHTML={
        {
          "__html": "Isabelle, known as Shizue in Japan, is a fictional character from the Animal Crossing series of video games. She is a kindly Shih Tzu that debuted in the 2012 release Animal Crossing: New Leaf, where she serves as the secretary to the player character.",
        }
      }
      hidden={true}
      id="basic--tabpanel-1"
      role="tabpanel"
      tabIndex={0}
    />
    <div
      aria-labelledby="basic--tab-2"
      className="chakra-tabs__tab-panel css-sjt5nk"
      dangerouslySetInnerHTML={
        {
          "__html": "<p>Totakeke, more commonly known as <b>K.K. Slider or K.K.</b>, is a fictional character within the Animal Crossing franchise. One of the franchise's most popular characters, he debuted in the title Animal Crossing, and has appeared in every installment since.</p>",
        }
      }
      hidden={true}
      id="basic--tabpanel-2"
      role="tabpanel"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Tabs renders the UI snapshot correctly 2`] = `
<div
  className="chakra-tabs css-kle7zy"
  id="chakra"
>
  <div
    className="css-l3f8vc"
  >
    <button
      aria-label="Scroll tabs left"
      className="chakra-button css-18qhlcl"
      id="tabs-previous-chakra"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="tabs-previous-icon-chakra"
        role="img"
        title="previous arrow"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="css-0"
    >
      <div
        className="css-tjdjds"
      >
        <div
          aria-orientation="horizontal"
          className="chakra-tabs__tablist css-k008qs"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="chakra--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="chakra--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Tom Nook
          </button>
          <button
            aria-controls="chakra--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="chakra--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Isabelle
          </button>
          <button
            aria-controls="chakra--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="chakra--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            K.K. Slider
          </button>
        </div>
      </div>
    </div>
    <button
      aria-label="Scroll tabs right"
      className="chakra-button css-1djleqc"
      id="tabs-next-chakra"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="tabs-next-icon-chakra"
        role="img"
        title="next arrow"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
  </div>
  <div
    className="chakra-tabs__tab-panels css-8atqhb"
  >
    <div
      aria-labelledby="chakra--tab-0"
      className="chakra-tabs__tab-panel css-sjt5nk"
      hidden={false}
      id="chakra--tabpanel-0"
      role="tabpanel"
      tabIndex={0}
    >
      <p>
        Tom Nook, 
        <b>
          known in Japan as Tanukichi
        </b>
        , is a fictional character in the Animal Crossing series who operates the village store.
      </p>
    </div>
    <div
      aria-labelledby="chakra--tab-1"
      className="chakra-tabs__tab-panel css-sjt5nk"
      dangerouslySetInnerHTML={
        {
          "__html": "Isabelle, known as Shizue in Japan, is a fictional character from the Animal Crossing series of video games. She is a kindly Shih Tzu that debuted in the 2012 release Animal Crossing: New Leaf, where she serves as the secretary to the player character.",
        }
      }
      hidden={true}
      id="chakra--tabpanel-1"
      role="tabpanel"
      tabIndex={0}
    />
    <div
      aria-labelledby="chakra--tab-2"
      className="chakra-tabs__tab-panel css-sjt5nk"
      dangerouslySetInnerHTML={
        {
          "__html": "<p>Totakeke, more commonly known as <b>K.K. Slider or K.K.</b>, is a fictional character within the Animal Crossing franchise. One of the franchise's most popular characters, he debuted in the title Animal Crossing, and has appeared in every installment since.</p>",
        }
      }
      hidden={true}
      id="chakra--tabpanel-2"
      role="tabpanel"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Tabs renders the UI snapshot correctly 3`] = `
<div
  className="chakra-tabs css-1xdhyk6"
  data-testid="props"
  id="props"
>
  <div
    className="css-l3f8vc"
  >
    <button
      aria-label="Scroll tabs left"
      className="chakra-button css-18qhlcl"
      id="tabs-previous-props"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="tabs-previous-icon-props"
        role="img"
        title="previous arrow"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="css-0"
    >
      <div
        className="css-tjdjds"
      >
        <div
          aria-orientation="horizontal"
          className="chakra-tabs__tablist css-k008qs"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="props--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="props--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Tom Nook
          </button>
          <button
            aria-controls="props--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="props--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Isabelle
          </button>
          <button
            aria-controls="props--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-p5yrdq"
            id="props--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            K.K. Slider
          </button>
        </div>
      </div>
    </div>
    <button
      aria-label="Scroll tabs right"
      className="chakra-button css-1djleqc"
      id="tabs-next-props"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="tabs-next-icon-props"
        role="img"
        title="next arrow"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
  </div>
  <div
    className="chakra-tabs__tab-panels css-8atqhb"
  >
    <div
      aria-labelledby="props--tab-0"
      className="chakra-tabs__tab-panel css-sjt5nk"
      hidden={false}
      id="props--tabpanel-0"
      role="tabpanel"
      tabIndex={0}
    >
      <p>
        Tom Nook, 
        <b>
          known in Japan as Tanukichi
        </b>
        , is a fictional character in the Animal Crossing series who operates the village store.
      </p>
    </div>
    <div
      aria-labelledby="props--tab-1"
      className="chakra-tabs__tab-panel css-sjt5nk"
      dangerouslySetInnerHTML={
        {
          "__html": "Isabelle, known as Shizue in Japan, is a fictional character from the Animal Crossing series of video games. She is a kindly Shih Tzu that debuted in the 2012 release Animal Crossing: New Leaf, where she serves as the secretary to the player character.",
        }
      }
      hidden={true}
      id="props--tabpanel-1"
      role="tabpanel"
      tabIndex={0}
    />
    <div
      aria-labelledby="props--tab-2"
      className="chakra-tabs__tab-panel css-sjt5nk"
      dangerouslySetInnerHTML={
        {
          "__html": "<p>Totakeke, more commonly known as <b>K.K. Slider or K.K.</b>, is a fictional character within the Animal Crossing franchise. One of the franchise's most popular characters, he debuted in the title Animal Crossing, and has appeared in every installment since.</p>",
        }
      }
      hidden={true}
      id="props--tabpanel-2"
      role="tabpanel"
      tabIndex={0}
    />
  </div>
</div>
`;
